### ISSUE TITLE
Missing PostgreSQL types in PL/Rust

### ISSUE THREAD

**Issue by jkatz** [tcdi/plrust#203]
The following types are available in pgx but missing in PL/Rust:

- [ ] The `timestamptz`, `date` etc. family of types
- [x] `uuid`
- [x] `point` and other geometric types
- [x] (Arrays) -- this may already work, have not tested.
- [x] (Ranges)
- [ ] `inet`
- [ ] `macaddr`
- [ ] `cidr`

The timestamp series + UUID are pretty commonly used and if it's a matter of choosing which ones to prioritize, I'd vote for those.

---

EDITED(workingjubilee): Added a few more that came up in conversation

**Comment by eeeebbbbrrrr** [comment]
A few things here:

- `point`, `box`: pgx has basic From/IntoDatum support for these, and their x,y/high,low members are public f64s, so that's fine for users.  Adding to plrust
- `uuid` has nice support in pgx.  Adding to plrust
- `inet`: pgx has From/IntoDatum support but it treats it as a String with no other infrastructure around the actual type.  This type needs a major rework in pgx.  **NOT** adding to plrust at this time
- `macaddr`, `cidr`: pgx does not have support for these at all.  Like `inet`, they'd need some major work in pgx first.  **NOT** adding to plrust at this time
- "time/date" types: these are already in plrust
- Arrays of any plrust-supported type is already there
- Ranges over i32, i64, AnyNumeric, Numeric, Date, Timestamp, TimestampwithTimeZone, are supported in pgx.  Adding to plrust

tl;dr:

## Already Supported
- date/time types
- Arrays over any other supported type

## Adding Support For
- point
- box
- uuid
- Ranges Ranges over i32, i64, AnyNumeric, Numeric, Date, Timestamp, TimestampwithTimeZone
 
## Not Supported For v1.0
- macaddr
- cidr
- inet

**Comment by eeeebbbbrrrr** [comment]
Note:  This is going to be contingent on releasing a pgx 0.7.2.  Unfortunately its `IntoDatum` impls for both `Point` and `BOX` are bugged.  I've PR'd that here:  https://github.com/tcdi/pgx/pull/1042

**Comment by eeeebbbbrrrr** [comment]
WIP PR:  https://github.com/tcdi/plrust/pull/213

**Comment by eeeebbbbrrrr** [comment]
Edit:  I edited the comments above.  I just realized that plrust only half-supports the date/time types.  Will resolve...
